acm/src/main/java/org/literacybridge/acm/gui/ResourceView/audioItems/AudioItemTableModel.java

```diff
@@ -2,57 +2,134 @@
 
 import java.io.File;
 import java.text.SimpleDateFormat;
+import java.util.Comparator;
 import java.util.Date;
 import java.util.List;
+import java.util.Map;
 
 import javax.swing.table.AbstractTableModel;
 
-import org.apache.commons.lang.StringUtils;
 import org.literacybridge.acm.config.ACMConfiguration;
 import org.literacybridge.acm.gui.Application;
+import org.literacybridge.acm.gui.ResourceView.audioItems.ColumnInfo.ValueProvider;
 import org.literacybridge.acm.gui.dialogs.audioItemPropertiesDialog.AudioItemPropertiesModel;
+import org.literacybridge.acm.gui.resourcebundle.LabelProvider;
 import org.literacybridge.acm.gui.util.AudioItemNode;
 import org.literacybridge.acm.gui.util.UIUtils;
 import org.literacybridge.acm.gui.util.language.LanguageUtil;
 import org.literacybridge.acm.repository.AudioItemRepository.AudioFormat;
 import org.literacybridge.acm.store.AudioItem;
+import org.literacybridge.acm.store.Committable;
 import org.literacybridge.acm.store.MetadataSpecification;
 import org.literacybridge.acm.store.MetadataStore;
-import org.literacybridge.acm.store.MetadataValue;
+import org.literacybridge.acm.store.MetadataStore.DataChangeListener;
 import org.literacybridge.acm.store.Playlist;
-import org.literacybridge.acm.store.SearchResult;
 
-public class AudioItemTableModel extends AbstractTableModel {
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
 
-    private static final long serialVersionUID = -2998511081572936717L;
+public class AudioItemTableModel extends AbstractTableModel implements DataChangeListener {
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
 
-    private SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+    public static final ColumnInfo<String> INFO_ICON_COLUMN = ColumnInfo.newColumnInfo("", 25, ColumnInfo.WIDTH_NOT_SET, new ValueProvider<String>(true) {
```

Comment:
The references are static final, but the objects themselves are not constant; they're not even immutable (the objects have setter methods, after all). They're the current state of the column, not the one-and-only state the column can ever have. Since they're not constants, I wouldn't make them upper case.